public/pages/workflow_detail/workflow_detail.test.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import {
+  RouteComponentProps,
+  Route,
+  Switch,
+  MemoryRouter,
+} from 'react-router-dom';
+import { WorkflowDetail } from './workflow_detail';
+import { WorkflowDetailRouterProps } from '../../pages';
+import '@testing-library/jest-dom';
+import { mockStore } from '../../../test/utils';
+
+jest.mock('../../services', () => {
+  const { mockCoreServices } = require('../../../test');
+  return {
+    ...jest.requireActual('../../services'),
+    ...mockCoreServices,
+  };
+});
+
+const renderWithRouter = (initialEntries: string[]) =>
+  render(
+    <Provider store={mockStore}>
+      <MemoryRouter initialEntries={initialEntries}>
```

Comment:
How does `MemoryRouter` work? What about doing something similar to AD, where we can pass in some limited info (resource ID for example), and construct the route inline? See https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/DetectorDetail/containers/__tests__/DetectorDetail.test.tsx#L33:L39 as an example.

Reply:
- Using MemoryRouter is better for tests because it lets you simulate routing in memory without needing a real URL, making tests easier and more controlled.
- https://v5.reactrouter.com/web/api/MemoryRouter

Reply:
- Alternate option is to  use`Router instead of MemoryRouter with  history = createMemoryHistory(); ` Will try this option and raise a PR. Thank you.

Reply:
Is there a use case we want/need to persist history? I'm only proposing to keep it simple how we have other tests and just use `Router`. We still are passing the url path on line 34 regardless.

Reply:
- @ohltyler,  We need history while rendering header component . https://github.com/saimedhi/dashboards-flow-framework/blob/main/public/pages/workflow_detail/components/header.tsx#L58. 

- In such cases, AD is using `createMemoryHistory();` in test cases. Let me know if you feel something is incorrect. 

- If history is not given getting below errors. 
```

The prop `history` is marked as required in `Router`, but its value is `undefined`.

Uncaught [TypeError: Cannot read properties of undefined (reading 'location')] 
          
          
```

Reply:
got it! That makes sense